### DISCUSSION TITLE
How to access param added via `register_action_param`

### DISCUSSION THREAD

**Discussion by rgstephens** [NVIDIA-NeMo/Guardrails#499] [Q&A] [answered]
Seems like this must be simple but the [docs](https://github.com/NVIDIA/NeMo-Guardrails/blob/65649b6658df589eb8dac182519482f18ef6ed84/docs/user_guides/configuration-guide.md?plain=1#L56) show an example of `register_action_param` called `db`.

Let's say I have an existing action called `start_round ` which already has a parameter called `user_id`. How do I access `db`?

```py
async def start_round(db, user_id, context: Optional[dict] = None):
```

If I define it as one of the parameters to the action like above, I get the error: `Error while execution start_round: start_round() missing 1 required positional argument: 'db'`

**Comment by drazvan** [comment, marked as answer]
Hi @rgstephens !

The parameters that are registered using `register_action_param` are passed automatically to the actions when they are invoked. You can think of them as keyword parameters with default values in Python. 

If you're getting the error above, it looks like the registration did not take place correctly. Can you share a minimal example to reproduce this. Thanks!

**Reply by rgstephens** [reply]
You're right, the registration wasn't happening becuase my `config.py` was in the `actions` directory instead of the `config` directory.
